lib/galaxy/tool_shed/util/hg_util.py

```diff
@@ -113,10 +113,8 @@ def get_repo_for_repository(app, repository=None, repo_path=None):
         hg,
         ui
     )
-    if repository is not None:
-        return hg.repository(ui.ui(), repository.repo_path(app))
-    if repo_path is not None:
-        return hg.repository(ui.ui(), repo_path)
+    repo_path = repo_path if repo_path else repository.repo_path(app)
```

Comment:
True, but you need to pass either repo_path or repository, so this still can't happen (because if `repo_path` is passed in you wouldn't evaluate the right side). This method's gone after https://github.com/galaxyproject/galaxy/pull/9580, so let me rebase that